internal/remoteconfig/remoteconfig.go

```diff
@@ -169,15 +170,28 @@ func (c *Client) updateState() {
 		resp.Body.Close()
 	}()
 
-	var update clientGetConfigsResponse
-	err = json.NewDecoder(resp.Body).Decode(&update)
+	if sc := resp.StatusCode; sc != http.StatusOK {
+		log.Debug("remoteconfig: http request error: response status code is not 200 (OK) but %s", http.StatusText(sc))
+		return
+	}
+
+	respBody, err := io.ReadAll(resp.Body)
 	if err != nil {
-		c.lastError = err
+		log.Error("remoteconfig: http request error: could not read the response body: %v", err)
 		return
 	}
 
-	err = c.applyUpdate(&update)
-	c.lastError = err
+	if body := string(respBody); body == `{}` || body == `null` {
+		return
+	}
```

Comment:
@Hellzy, on top of the issues I fixed in this PR, what you see here is the reason why we got the json parsing errors being reported to the backend: `{}` is the "no update" payload and `repository.Update(update)` doesn't handle this case properly yet.

We will be able to avoid and remove this check once @ameske releases a new version of `Update()` properly handling this case without returning an error.

Reply:
Thanks for looking into this :+1: